step-templates/put-server-back-in-azure-load-balancer.json

```diff
@@ -1,68 +1,84 @@
-  "Description": "Uses Service Principal to Put Server Back In Azure Load Balancer.",
-  "ActionType": "Octopus.AzurePowerShell",
-  "Version": 1,
-  "CommunityActionTemplateId": null,
-  "Properties": {
-    "Octopus.Action.Script.ScriptSource": "Inline",
-    "Octopus.Action.Script.ScriptBody": "#region Verify variables\n\n#Verify psbilbAzureSubscription is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureSubscription']))\n{\n  Throw 'Azure Subscription cannot be null.'\n}\n$azureSubscription = $OctopusParameters['psbilbAzureSubscription']\nWrite-Host ('Azure Subscription: ' + $azureSubscription)\n\n#Verify psbilbAzureResourceGroup is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureResourceGroup']))\n{\n  Throw 'Azure Resource Group cannot be null.'\n}\n$azureResourceGroup = $OctopusParameters['psbilbAzureResourceGroup']\nWrite-Host ('Azure Resource Group: ' + $azureResourceGroup)\n\n\n#Verify psbilbAzureMachineName is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureMachineName']))\n{\n  Throw 'Azure Machine Name cannot be null.'\n}\n$azureMachineName = $OctopusParameters['psbilbAzureMachineName']\nWrite-Host ('Azure Machine Name: ' + $azureMachineName)\n\n#Verify psbilbAzureLoadBalancer is not null.\nIf ([string]::IsNullOrEmpty($OctopusParameters['psbilbAzureLoadBalancer']))\n{\n  Throw 'Azure Load Balancer cannot be null.'\n}\n$azureLoadBalancer = $OctopusParameters['psbilbAzureLoadBalancer']\nWrite-Host ('Azure Load Balancer: ' + $azureLoadBalancer)\n\n#endregion\n\n\n#region Process\n\nSet-AzureRmContext -SubscriptionName $azureSubscription\n\n$azureVM = Get-AzureRmVM -ResourceGroupName $azureResourceGroup -Name $azureMachineName\nIf (!$azureVM)\n{\n  Throw 'Could not retrieve server from Azure needed to remove from Load Balancer.'\n}\n\n$nic = (Get-AzureRmNetworkInterface -ResourceGroupName $azureResourceGroup | Where-Object {$_.VirtualMachine.Id -eq $azureVM.Id})\nIf (!$nic)\n{\n  Throw 'Could not retrieve NIC from Azure needed to remove from Load Balancer.'\n}\n\n$loadBalancer = Get-AzureRmLoadBalancer -Name $azureLoadBalancer -ResourceGroupName $azureResourceGroup\nIf (!$loadBalancer)\n{\n  Throw 'Could not retrieve Load Balancer info from Azure.'\n}\n\n$nic.IpConfigurations[0].LoadBalancerBackendAddressPools = $loadBalancer.BackendAddressPools\nSet-AzureRmNetworkInterface -NetworkInterface $nic\n\n#endregion\n",
-    "Octopus.Action.Package.FeedId": null,
-    "Octopus.Action.Script.ScriptFileName": null,
-    "Octopus.Action.Package.PackageId": null
-  },
-  "Parameters": [
-    {
-      "Id": "c967d075-8754-4131-8de0-9ff531a9c0e6",
-      "Name": "psbilbAzureSubscription",
-      "Label": "Azure Subscription",
-      "HelpText": null,
-      "DefaultValue": "",
-      "DisplaySettings": {
-        "Octopus.ControlType": "SingleLineText"
-      },
-      "Links": {}
+  {
+    "Id": "540de6f6-f1c6-4e1c-9062-9e78570074f6",
+    "Name": "Add Server to Azure Load Balancer",
```

Comment:
Could we keep the name as previous e.g. `Put Server Back In Azure Load Balancer` - this helps to prevent any sync issues with anyone who has a version of this.

Reply:
I kind of hate "Put Server Back In Azure Load Balancer" as it's not indicative of what the step actually does, the Azure VM doesn't have to be on the Load Balancer for it to work, you could add a brand new server to the LB for example.

Reply:
Let me see if the name change causes any sync issues - if it's done by id it is probably fine

Reply:
@benjimac93 - thanks for the patience, keep the new name, the sync is done by ID and not name, therefore you can leave your new name. However, could you rename the file to `add-server-to-azure-load-balancer.json` to keep the new name consistent :smile: